demo/ZeroToHeroUsingTheBasis.md

```diff
@@ -0,0 +1,357 @@
+# Configurations
+
+```
+version: "3"
+services:
+diwala-agent:
+image: bcgovimages/aries-cloudagent:py36-1.15-0_0.6.0rc0
+environment:
+ADMIN_PORT: 5000
+AGENT_PORT: 10000
+INDY_SEED: diwalaJnecnKjegsyIR0P3vXBc56G71H
+WEBHOOK_URL: http://host.docker.internal:3000/webhooks
+AGENT_LABEL: Diwala Agent
+ports:
+- "5000:5000"
+- "10000:10000"
+entrypoint: /bin/bash
+command: [
+"-c",
+"curl -d '{\"seed\":\"diwalaJnecnKjegsyIR0P3vXBc56G71H\", \"role\":\"TRUST_ANCHOR\", \"alias\":\"Diwala agent\"}' -X POST http://test.bcovrin.vonx.io/register; \
+sleep 5; \
+aca-py start \
+-it http '0.0.0.0' 10000 \
+-ot http \
+--admin '0.0.0.0' 5000 \
+-e 'http://localhost:10000/' \
+--auto-provision \
+--wallet-storage-type postgres_storage \
+--wallet-storage-config '{\"url\":\"host.docker.internal:5432\", \"wallet_scheme\":\"MultiWalletSingleTableSharedPool\"}' \
+--wallet-storage-creds '{\"account\":\"local-cloud-agent\",\"password\":\"testtest\",\"admin_account\":\"admin-cloud-agent\",\"admin_password\":\"testtest\"}' \
+--wallet-type indy \
+--wallet-name diwala \
+--wallet-key Diwala.Agent321311 \
+--seed diwalaJnecnKjegsyIR0P3vXBc56G71H \
+--genesis-url http://test.bcovrin.vonx.io/genesis \
+--label 'Diwala Multitenant Agent' \
+--admin-insecure-mode \
+--multitenant \
+--multitenant-admin \
+--jwt-secret very_secret_secret \
+--log-level info",
+]
+
+```
+
+## Postgres
+Postgres connection is important to make sure you are keeping state in a sane place. Volumes are fine, but they can be lost quite fast if it is not a good control over them. So a database, potentially as a service is even easier to handle and make sure that have the backups needed and overview needed.
+
+Setting up configuration for postgres is quite straight forward, with one hickup. The above configuration does not work out of the box, atleast not on < postgres@12.
+
+The setup fails because it cannot find the used “user” database. Which is weird, as the user is not a database, that is created during initialisation.
+
+To get past this, I had to create a “fake” database with the “user” name and it was able to run initialisation and all for the rest of the indy database creation.
+
+### Database management mode
+It is important to be aware of what is default and what possibilities you have for database management mode. Here is an explainer: https://github.com/hyperledger/indy-sdk/tree/master/experimental/plugins/postgres_storage#wallet-management-modes
+I chose MultiWalletSingleTableSharedPool because centralise all the things 🤪
+
+### Questions
+Is there any way to read out of the database and understand the binary data? Meaning setting up statistics towards the type column of a wallet db?
```

Comment:
With the Indy wallet setup, this is not possible -- all of the data is encrypted and there is not an easy way around that. The new storage mechanism that is coming using Aries-Askar will have the capability of having a "null" encryption method. This is a security trade-off -- being able to create better code as a developer vs. the risk of using null encryption in production.

Reply:
Adding this info here: https://github.com/hyperledger/aries-cloudagent-python/pull/1004/commits/284ac632e1dbb7810634fdcc2149755c4792dbca

https://github.com/hyperledger/aries-cloudagent-python/pull/1004/commits/3b3223ed0f097983d1a88722b5bc91cd10fa8c11